subscription_worker.go

```diff
@@ -22,14 +22,15 @@ type SubscriptionWorker struct {
 
 	cancellationRequested int32 // atomic, > 0 means cancellation was requested
 	options               *SubscriptionWorkerOptions
-	subscriber            func(*SubscriptionBatch) error
+	chResults 			chan []*SubscriptionBatchItem
```

Comment:
I don't think we want this. We want a channel of `subscriptionBatch`, not invididual `SubscriptionBatchItem`
There is interesting information about the batch as well (includes, the ability to open a session from it, etc.

Reply:
Ok, will look into making it work.

The reason I made this change is that in java code there's only one `SubscriptionBatch` for the lifetime of the worker so it's being modified by the worker while it's being read by the client, which is inherently unsafe (namely, race detector flagged changing `Items` property by the worker).

I considered creating a `SubscriptionBatch` for each batch but it looks like a havey-weight object.

To make this work I'll probably rename current `SubscripttionBatch` to something else and introduce  a `SubscriptionBatch` that provides the same information as current `SubcriptionBatch` but is immutable and therefore safe to acess in the client. Or just send a lightweight, data only copy of `SubscriptionBatch`.

Reply:
The idea is that the batch is modified by both the callback and the subscription, but never together.
The subscription modify the batch instance between invocations of the callback. Obviously not going to work with a channel, so your approach sounds good